### DISCUSSION TITLE
BlueSCSI issue with PowerBook 550c

### DISCUSSION THREAD

**Discussion by mac68kuser** [BlueSCSI/BlueSCSI-v2#162] [Help]
Hello,
A BlueSCSI PowerBook edition installed in a PB 550c will not reset properly at system shutdown or reboot if an external device that provides termination power is connected and powered up. Duplicate by installing a BlueSCSI device, then connect a CD-ROM or other external device that provides SCSI termination power (termination power is required for external SCSI devices). The system will boot, but after a shutdown or reboot, the dreaded disk with a question mark appears. Cycling power on the external device while the PB 550c is shutdown is a workaround. Apparently a device that provides termination power keeps the internal BlueSCSI alive after a system shutdown, and whatever process the PB 550c uses at startup to reset the SCSI bus (in Mac OS 8.1) isn't enough to reset the BlueSCSI device.
-Stan

**Comment by erichelgeson** [comment]
Hi there - I'll give this a test on my 540c and see if I can reproduce the issue. The BlueSCSI should just be picked up on the reboot.

**Reply by erichelgeson** [reply]
Just tested on my 540c and I cant reproduce this - installed 8.1. was able to reboot and power off and on all while the bluescsi was still on.

Could you test this again, but this time power the BlueSCSI via USB? Maybe the term power is dipping or unstable and gets the bluescsi in an odd state. Check the logs too.

**Comment by mac68kuser** [comment]
Hello Eric,

Thank you for your response. I did a little more checking on this. Normal shutdowns and reboots seem to work ok. I have an external CD-ROM drive attached (configured to provide termination power), and a SCSI2SD v5.0b also attached, as follows:

BlueSCSI Internal (terminated) <-> PB 550c <-> CD-ROM <-> SCSI2SD (terminated)

You should be able to duplicate by booting a Linux kernel (if you have no Linux root filesystem, don't include "root=/dev/xyz" and the kernel will simply crash). Then reset the system using control-command-power, and the flashing disk icon will eventually come up. If you then unplug the system, plug it back in and restart, it also can't find the startup disk (the internal BlueSCSI). The only way to fix it is to turn off the CD-ROM drive and the computer, then turn the CD-ROM drive on and then turn the computer on. I noticed that while the CD-ROM drive is on, the SCSI2SD is also on, though I haven't opened the PowerBook to see whether the BlueSCSI also remains on while the PowerBook is off due to the CD-ROM drive being on and supplying termination power.

If you're not able to boot Linux, you may also be able to duplicate the issue by having a CD-ROM supplying termination power connected, then in Mac OS unplug the system (more risky since Mac OS could become corrupted). Plug the system back in and power up while the CD-ROM drive is still on. This should result in the flashing disk icon.

It would be difficult to connect USB since this is the BlueSCSI PowerBook edition, though I could take the microSD card out and look at the log file.

-Stan

**Reply by erichelgeson** [reply]
You're going to have to explain your setup a bit more... when did linux come into play :D - If you wanted to zip up your entire contents of your SD card so I have the exact same setup and share it if possible (can send it privately too, let me know)

Please explain a bit more on the procedure that leads to the unexpected state. If it's a crash or some other oddness outside scsi, that might be difficult for us to resolve. Either way we'll need more info.

Luckily the keyboard/drive on the 500 series is pretty easy to leave unscrewed while testing.

**Comment by mac68kuser** [comment]
Linux isn't important, it just happened to be where I first noticed the problem. Any forced soft or hard reset with an external device connected should trigger the problem.

BlueSCSI PB edition <-> PB500c <-> CD-ROM drive

The BlueSCSI and CD-ROM drive are terminated. The CD-ROM drive is providing termination power (devices on the external SCSI connection must provide termination power). My PB 550c has a dead battery and a dead (or removed) PRAM battery.

If you boot Mac OS from the BlueSCSI, with any device that provides termination power connected to the external SCSI, then I believe you should be able to duplicate the issue with a soft reset (control-command-power) or pull the power plug (assumes no working battery is installed). When you reboot the system, with the external device still on, it displays the flashing disk icon.

Without an external device connected and turned on, I'm not able to reproduce the error.

To fix, turn everything off, including the external device(s), then turn the external device(s) on, then reboot the PB 550c. I suspect the external device that is providing termination power is keeping the BlueSCSI powered up when the PB 550c resets or loses power, and the BlueSCSI is not responding as it should to the SCSI reset that the PB 550c issues when it powers up.

**Comment by mac68kuser** [comment]
I just tried pulling the power plug and doing a soft reset, and I wasn't able to duplicate the problem this morning.

So nevermind, and I'm sorry for wasting time and bandwidth. If I discover how to duplicate the problem, I will let you know.

thanks

**Reply by erichelgeson** [reply]
No need to apologize, it's not a waste of bandwidth to help :) - if you can get to a minimal reproducible example I have my 540c here to help.
